### DISCUSSION TITLE
Auto updater to save/skip updating user data or preferences?

### DISCUSSION THREAD

**Discussion by phantomdenkali** [tauri-apps/tauri#7102] [Q&A] [answered]
Hi, I'm new to Tauri and it looks great. I've been looking into the auto-updater feature, and while useful, I'm not certain if there is a way to skip updating user preference files or directories. 

I was also hoping to use Tauri to package and run an express server bundled with the management ui.  There was a nice thread and work-around posted [here](https://github.com/tauri-apps/tauri/discussions/2613#discussioncomment-5893831) but if there are files that I do not want to upgrade or save as user-preferences I'm not certain if the updated binary will wipe them clean.

**Comment by FabianLars** [comment, marked as answer]
The updater will only touch the files in the installation directory, so whatever you don't want it to touch you can store in user data directories like https://tauri.app/v1/api/js/path#appconfigdir for example.

**Comment by phantomdenkali** [comment]
That's great. Are there pre/post install & update hooks to run scripts for the update. I'm thinking all the binaries can be kept in the installation directory, and the settings/configurations/data can be stored outside. On update, can setup/prep/shutdown running services, and copy the binaries out to use the extant user data.

**Reply by FabianLars** [reply]
Sorry for the delay, i think i messed up my github notification settings 😅 

With the updater behaving pretty different per platform (and the app being closed while updating on some/most? of them) there aren't really any good hooks for when the app _was_ updated so you'd have to DIY that somehow (idk, by saving the last version somewhere and comparing it with the currently running instance?).
Doing something _before_ the update is installed is no problem.

This is something i would love to see better implemented but with the platform&bundle differences that is _really_ tricky.

**Reply by phantomdenkali** [reply]
No worries, thanks for the reply.

"Doing something before the update is installed is no problem." -- how do you do that? Perhaps I can copy the configurations out to a backup file before, and check on that at startup every time in the app itself.

**Reply by FabianLars** [reply]
You'd have to disable tauri's default updater dialog (iirc will be removed for v2 anyway) and then you'll have access to events and control over when it's installed: https://tauri.app/v1/guides/distribution/updater/#custom-dialog.

~~Though, reading through your initial question again, i'm not sure if that will actually help you much, considering that the installdir (including the sidecar/resource location) is read-only~~
